src/casl/casl-ability.factory.ts

```diff
@@ -98,17 +98,34 @@ export class CaslAbilityFactory {
       /*  unauthenticated users
       **/
 
+      can(Action.DatasetReadManyPublic, DatasetClass);
+      can(Action.DatasetReadOnePublic, DatasetClass, {
+        isPublished: true,
+      });
```

Comment:
Do we need separated DatasetReadPublic permissions? One DatasetReadPublicAny feels enough for me. 
We can discuss with Max about that later for the final merge into master